packages/core/src/internal/middleware.js

```diff
@@ -1,11 +1,10 @@
 import * as is from '@redux-saga/is'
 import { check, assignWithSymbols, createSetContextWarning } from './utils'
 import { stdChannel } from './channel'
-import { identity } from './utils'
 import { runSaga } from './runSaga'
 
-export default function sagaMiddlewareFactory({ context = {}, ...options } = {}) {
-  const { sagaMonitor, logger, onError, effectMiddlewares } = options
+export default function sagaMiddlewareFactory(options = {}) {
+  const { context = {}, channel = stdChannel(), sagaMonitor, logger, onError, effectMiddlewares } = options
```

Comment:
I would inline this destructuring into `sagaMiddlewareFactory` declaration, without it I'm thinking that `options` might be used later on and search for references to it
```js
export default function sagaMiddlewareFactory({
	channel = stdChannel(),
	context = {},
	effectMiddlewares
	logger,
	onError,
	sagaMonitor,
} = {}) {
	// ...
}
```